packages/venia-ui/lib/util/makeUrl.js

```diff
@@ -1,8 +1,21 @@
+import { BrowserPersistence } from '@magento/peregrine/lib/util';
+
+const storage = new BrowserPersistence();
+
 // If the root template supplies the backend URL at runtime, use it directly
 const htmlDataset = document.querySelector('html').dataset;
 const { imageOptimizingOrigin } = htmlDataset;
 // Protect against potential falsy values for `mediaBackend`.
-let mediaBackend = htmlDataset.mediaBackend;
+const storeCode = storage.getItem('store_view_code') || STORE_VIEW_CODE;
+const storeSecureBaseMediaUrl = {};
+
+AVAILABLE_STORE_VIEWS.forEach(store => {
```

Comment:
Where does `AVAILABLE_STORE_VIEWS` come from? A comment here would be helpful for the future.